wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitBasicAuthenticationCredentialsProvider.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.git;
+
+import com.google.inject.Singleton;
+import org.eclipse.che.api.git.exception.GitException;
+import org.eclipse.che.api.git.shared.ProviderInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Credentials provider for Git basic authentication
+ *
+ * @author Yossi Balan
+ */
+@Singleton
+public class GitBasicAuthenticationCredentialsProvider implements CredentialsProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GitBasicAuthenticationCredentialsProvider.class);
+    private static ThreadLocal<UserCredential> currRequestCredentials = new ThreadLocal<>();
+    private static final String GIT_PROJECT_ID = "git-project-id";
```

Comment:
Are you sure that `git-project-id` is good name for CredentialsProvider? Mb `git-basic` or just `basic` would be better?